### PR TITLE
[Validator] Allow `ConstraintViolation::__toString()` to expose codes that are not null or emtpy strings

### PR DESCRIPTION
**Pull request by phansys**
|Q            |A  |
|---          |---|
|Branch       |2.8|
|Bug fix?     |yes|
|New feature? |no |
|BC breaks?   |no |
|Deprecations?|no |
|Tests pass?  |yes|
|Fixed tickets|n/a|
|License      |MIT|
|Doc PR       |n/a|
                   

Allow to expose `0` or `"0"` validation codes.

### CODE REVIEW

File: src/Symfony/Component/Validator/ConstraintViolation.php [hunk 1]
```diff
@@ -79,13 +79,13 @@ public function __toString()
         }
 
         $propertyPath = (string) $this->propertyPath;
-        $code = $this->code;
+        $code = trim($this->code);
```
Comment (ro0NL) [inline src/Symfony/Component/Validator/ConstraintViolation.php]: trimming is a new behavior, which IMHO is not needed.

Comment (phansys) [reply src/Symfony/Component/Validator/ConstraintViolation.php]: I understand your point, and I agree. My concern was originated by the absence of delimiters around the printed code, so I thought that readability could be improved trimming any undesired blankspace around the code.

Comment (ro0NL) [reply src/Symfony/Component/Validator/ConstraintViolation.php]: the delimiter is `(code ` and `)` :)

> undesired blankspace

we don't know that

**Review by ro0NL — APPROVED** [review]

File: src/Symfony/Component/Validator/ConstraintViolation.php (hunk 1 above)
Comment (phansys) [reply src/Symfony/Component/Validator/ConstraintViolation.php]: I mean cases like
```
(code    )
```
I think it is not clear which space comes from the dumped message and which one comes from the actual code, but I'm pretty sure this is an edge case that we can ignore.

**Review by nicolas-grekas — APPROVED** [review]
(for 3.4 as 2.8 is EOLed now)

**Comment by phansys** [comment]
Updated base to target 3.4.

**Pull request merged**